### ISSUE TITLE
--merge-pr reporting "test suite passes: pending... => not eligible for merging!" even though all tests seem to have passed

### ISSUE THREAD

**Issue by migueldiascosta** [easybuilders/easybuild-framework#3490]
further inspection showed that GitHub is reporting two tests, 'https://github.com/apps/dependabot' and 'https://github.com/apps/hound', to have conclusion `None` instead of `Success`

this was in a PR to master, might be specific to this branch due to GitHub policies

follow up with GitHub issue?

**Comment by migueldiascosta** [comment]
fwiw, this is still affecting https://github.com/easybuilders/easybuild-framework/pull/3526 and https://github.com/easybuilders/easybuild-easyblocks/pull/2277, at least (again, PRs to master)

**Comment by boegel** [comment]
For #3526, see https://api.github.com/repos/easybuilders/easybuild-framework/commits/51535c5e131115fdaff06353e144f8228be05349/check-suites .

There's 3 "check suites" there that are stuck in `status: queued` and `conclusion: null`, for `dependabot`, `hound` and `travis-ci` (see `app.slug`).

The latter happens because we've run out of free credits in Travis CI, but we'll get rid of that by simply removing the `.travis.yml` configuration files going forward (which means we won't be tested with Python 2.6 anymore, but we've deprecated support for that a while ago anyway).

For the other two (`dependabot` and `hound`), I think we'll need to actively filter those out and simply ignore the results in `--merge-pr` for PRs to `master`, we can't just disable those checks somehow without having unwanted side affects (like not checking other branches, or totally disabling `dependabot`).

**Comment by boegel** [comment]
@migueldiascosta Do we need to keep this open, after #3604 and #3607 ?

**Comment by migueldiascosta** [comment]
@boegel I think this is a separate issue but I´ll check

**Comment by migueldiascosta** [comment]
@boegel yeah, separate issue, and problem is still there. We may want to add something like

```diff
--- a/easybuild/tools/github.py
+++ b/easybuild/tools/github.py
@@ -2274,6 +2274,7 @@ def det_commit_status(account, repo, commit_sha, github_user):
 
         # if any check suite hasn't completed yet, final result is still pending
         if status in ['queued', 'in_progress']:
+            print("%s (%s): %s" % (check_suite_data['app']['name'], check_suite_data['url'], check_suite_data['status']))
             result = STATUS_PENDING
 
         # if check suite is completed, take the conclusion into account
```

which in the case of https://github.com/easybuilders/easybuild-framework/pull/3640 outputted:

```
Dependabot (https://api.github.com/repos/easybuilders/easybuild-framework/check-suites/2443283009): queued
Hound (https://api.github.com/repos/easybuilders/easybuild-framework/check-suites/2443283011): queued
Travis CI (https://api.github.com/repos/easybuilders/easybuild-framework/check-suites/2443283013): queued
```

**Comment by verdurin** [comment]
Seeing this at https://github.com/easybuilders/easybuild-easyconfigs/pull/14950 too

**Comment by verdurin** [comment]
https://gist.github.com/ce114b85f7457ed17809370c5770b8c6

**Comment by branfosj** [comment]
I think we need to check `status_last_commit` instead of (or as well as) `mergeable_state` of the latest check.

14950:
```
mergeable_state: clean
status_last_commit: failure
```

14963
```
mergeable_state clean
status_last_commit success
```
